packages/react-components/src/components/VideoGallery.tsx

```diff
@@ -192,6 +196,7 @@ export const VideoGallery = (props: VideoGalleryProps): JSX.Element => {
   const strings = { ...localeStrings, ...props.strings };
 
   const shouldFloatLocalVideo = !!(layout === 'floatingLocalVideo' && remoteParticipants.length > 0);
+  const shouldAddControlsLocalVideo = shouldAddControlsToLocalVideoTrampoline(layout, remoteParticipants);
```

Comment:
would be good to encapsulate `shouldFloatLocalVideo` and `shouldAddControlsLocalVideo` into an enum as they are mutually exclusive.
something like: `const localTileHost = 'grid' | 'floatingDraggable' | 'floatingStatic'`

Reply:
So looking into this refactor, I will make it in a seperate PR since the styles all depend on the boolean evaluation of shouldFloatLocalVideo. the style functions will need to be updated to support the new logic if we change this.

Reply:
OK with a separate PR. This becomes especially important in the light of my comment on changing the API -- we'll have two different props that are related to each other. It's best to merge them into a single enum as soon as possible in the implementation.